### PR TITLE
Sonar Pulse Implementation. Close #46.

### LINKED ISSUE #46: Implement the sonar pulse ability. User Story #1
This feature will take two days to implement

### PR DESCRIPTION
**Pull request by orourtim**

### CODE REVIEW

File: src/main/java/cs361/battleships/models/Board.java [hunk 1]
```diff
@@ -69,6 +69,94 @@ public Result attack(int x, char y) {
...
+				}
+			}
+		}
+		return false;
+	}
+
+	private void addOccupiedResultToResultList(int x, char y) {
+		Result temp = new Result();
+		temp.setResult(AtackStatus.OCCUPIED);
+		temp.setLocation(new Square(x, y));
```
Comment (Gerasheo) [inline src/main/java/cs361/battleships/models/Board.java]: Part of the contract for our results is to have each result contain the ship that it is referring to. The ship being detected should be added to the occupied result here.

File: src/test/java/cs361/battleships/models/BoardTest.java [hunk 2]
```diff
@@ -210,4 +211,29 @@ public void testAttackSurrender() {
...
+        List<Result> list = board.getAttacks();
+        Iterator iterator = list.iterator();
+        Result firstResult = (Result) iterator.next();
+        Result secondResult = (Result) iterator.next();
+        Result thirdResult = (Result) iterator.next();
+        assertEquals(3, list.size());
+        assertEquals(OCCUPIED, firstResult.getResult());
+        assertEquals(OCCUPIED, secondResult.getResult());
+        assertEquals(OCCUPIED, thirdResult.getResult());
+    }
```
Comment (Gerasheo) [inline src/test/java/cs361/battleships/models/BoardTest.java]: Maybe also confirm that the list doesn't have a next result after you go through those three?

File: src/main/java/cs361/battleships/models/Board.java (hunk 1 above)
Comment (sellenth) [reply src/main/java/cs361/battleships/models/Board.java]: good catch Jeremy

File: src/main/java/cs361/battleships/models/Game.java [hunk 3]
```diff
@@ -56,6 +56,16 @@ public boolean attack(int x, char  y) {
         return true;
     }
 
+    public boolean sonarPulse(int x, char y) {
+        if (opponentsBoard.sonarPulse(x, y)) {
```
Comment (sellenth) [inline src/main/java/cs361/battleships/models/Game.java]: You could just have this method return opponentsBoard.sonarPulse(x, y), it would be equivalent

**Pull request merged**